### DISCUSSION TITLE
Can you confirm, and state clearly in the docs, that sampling from a Replay buffer is "Sampling With Replacement"

### DISCUSSION THREAD

**Discussion by jamartinh** [ymd-h/cpprb#33] [Q&A]
Can you confirm, and state clearly in the docs, that sampling from a Replay buffer is "Sampling With Replacement"?

**Comment by jamartinh** [comment]
It is said in the doc of the sample method in the class reference. However due to high significance, for statistical and theoretical properties,  it is perhaps desirable to state it more clearly as a feature in the main docs.

**Reply by ymd-h** [reply]
@jamartinh 
Thank you for your feedback.

A transaction can be contained multiple times in the same mini-batch. (aka. "Sample with Replacement")
This can be tested with small buffer with larger batch size sampling.

`ReplayBuffer` samples transactions independently.
https://github.com/ymd-h/cpprb/blob/0513b058619d599ace1a0059ec7501bbf7a2be5a/cpprb/PyReplayBuffer.pyx#L1384

`PrioritizedReplayBuffer` uses stratified sampling, so that transactions in the same mini-batch are not completely independent.
https://ymd_h.gitlab.io/cpprb/features/per/#technical-detail
